src/App.tsx

```diff
@@ -15,71 +15,94 @@ export const goods = [
+  state: Readonly<State> = {
+    selectedGood: 'Jam',
+    className: 'has-background-success-light',
+    blueLight: 'is-info',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  render() {
+    const {
+      selectedGood,
+      className,
+      blueLight,
+    } = this.state;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    return (
+      <main className="section container">
+        {selectedGood === ''
+          ? <h1 className="title">No goods selected</h1>
+          : (
+            <h1 className="title is-flex is-align-items-center">
+              {`${selectedGood} is selected`}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                data-cy="ClearButton"
+                type="button"
+                className="delete ml-3"
+                onClick={() => this.setState(
+                  { selectedGood: '' },
+                )}
+              />
+            </h1>
+          )}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                data-cy="Good"
+                className={
+                  selectedGood === good
+                    ? className
+                    : ''
+                }
+              >
+                <td>
+                  <button
+                    data-cy={
+                      selectedGood === good
+                        ? 'RemoveButton'
+                        : 'AddButton'
+                    }
+                    type="button"
+                    className={
+                      `button ${
+                        selectedGood === good
+                          ? blueLight
+                          : ''
+                      }`
+                    }
+                    onClick={() => (
+                      selectedGood === good
+                        ? this.setState({ selectedGood: '' })
+                        : this.setState({
+                          selectedGood: good,
+                          className: 'has-background-success-light',
+                          blueLight: 'is-info',
```

Comment:
in this case, you don't have to write it because in your code you have never changed `className` and `blueLight` in your state